src/sentry/remote_config/docs/protocol.md

```diff
@@ -22,17 +22,66 @@ Retrieve a project's configuration.
 
 **Feature Object**
 
-| Field | Type   | Description                        |
-| ----- | ------ | ---------------------------------- |
-| key   | string | The name used to lookup a feature. |
-| value | any    | A JSON value.                      |
+| Field    | Type           | Description                                                                     |
+| -------- | -------------- | ------------------------------------------------------------------------------- |
+| key      | string         | The name used to lookup a feature.                                              |
+| value    | any            | A JSON value.                                                                   |
+| variants | array[Variant] | Features can have any number of possible variants which alter the return value. |
```

Comment:
```suggestion
| variants | union[array[Variant], null] | Features can have any number of possible variants which alter the return value. |
```

Reply:
Let's just return an empty array.  Why bother having a conditional check for null if we can avoid it?

Reply:
Same concern for rules and rollout.

Reply:
fair enough